### PR TITLE
Additional run-time configurations for Kernel Gateway

### LINKED ISSUE #265: Make KernelSpecManager and KernelManager configurable
We want to make KernelSpecManager and kernel_manager_class configurable at run-time. This gives users flexibility to customize how Kernelspecs are fetched and how kernels are started.

### PR DESCRIPTION
**Pull request by shiti-saxena**
resolves #265

### CODE REVIEW

File: kernel_gateway/gatewayapp.py [hunk 1]
```diff
@@ -277,6 +279,27 @@ def keyfile_default(self):
...
+    kernel_spec_manager = Instance(KernelSpecManager, allow_none=True)
+
+    kernel_spec_manager_class = Type(
+        default_value=KernelSpecManager,
+        config=True,
+        help="""
+        The kernel spec manager class to use. Should be a subclass
+        of `jupyter_client.kernelspec.KernelSpecManager`.
+
+        The Api of KernelSpecManager is provisional and might change
```
Comment (rolweber) [inline kernel_gateway/gatewayapp.py]: We shouldn't have API warnings from Jupyter Client in the kernel gateway code, I think.

Comment (shiti-saxena) [reply kernel_gateway/gatewayapp.py]: I wasn't sure what the format for help would be so I copied it from [Jupyter Notebook](https://github.com/jupyter/notebook/blob/master/notebook/notebookapp.py#L997-L1007)

**Comment by shiti-saxena** [comment]
@rolweber I've updated the PR with the suggested changes. Please review it.

**Comment by parente** [comment]
The config options look reasonable to me for advanced users. Anyone looking to add a custom kernel manager will need to be careful to support all the mode and option behaviors of the default managers.

**Pull request merged**